java/datastructures/lib/src/main/java/codechallenges/linkedlistinsertion/linkedlistinsertions.java

```diff
@@ -0,0 +1,80 @@
+// append
+public void append(int value) {
+//create new node
+  Node newTail = new Node(value);
+//make tail point at new node
+  this.tail.next = newTail;
+  this.tail = newTail;
+  }
+
+public void append(int value) {
+// create new node for tail
+  Node newTail = new Node(value)
+// create a temp/curr node for traversal
+  Node curr = this.head;
+// traverse the list (stop at last node)
+  while(curr.next != null) {
+  curr = curr.next;
+  }
+// make the node stopped at point at new tail
+  curr.next = newTail
+  }
+
+public void insertBefore(int value, int newValue) throw illegalArgumentException {
```

Comment:
Watch your capitalization here. You want to invoke the `IllegalArgumentException` class, and classes are always capitalized. Otherwise, Java will think you're trying to reference a variable called `illegalArgumentException` and will throw an error.